solutions/silver/cf-1000C.mdx

```diff
@@ -63,31 +62,181 @@ int main() {
 		coords.push_back(c);
 		cur++;
 	}
-	vi freq(2 * n);  // This stores the frequency of a given endpoint.
+	// This stores the frequency of a given endpoint.
+	vector<int> freq(2 * n, 0);
 	for (int i = 0; i < n; i++) {
 		/*
 		 * A segment starts from [l*, r*], so we should end the segment
 		 * at r* + 1 to ensure that r* is included into the segment.
 		 */
-		freq[compressed[segments[i].f]]++;
-		freq[compressed[segments[i].s + 1]]--;
+		freq[compressed[segments[i].first]]++;
+		freq[compressed[segments[i].second + 1]]--;
 	}
-	// This counts how many points are at a given compressed endpoint using
-	// prefix sums.
+	// prefix sums for number of points at a given compressed endpoint
 	for (int i = 1; i < 2 * n; i++) { freq[i] += freq[i - 1]; }
 
-	vector<ll> cnt(n + 1);
+	// cnt[i] := number of points covered by exactly i segments
+	vector<ll> cnt(n + 1, 0);
 	for (int i = 1; i < coords.size(); i++) {
 		/*
-		 * We add r - l to the number of points for a given
-		 * frequency since we compressed the coordinates.
+		 * We decompress the endpoints for a given frequency and add the points
+		 * in this range to the answer.
 		 */
 		cnt[freq[i - 1]] += coords[i] - coords[i - 1];
 	}
 
-	for (int i = 1; i <= n; i++) {
-		cout << cnt[i] << " ";  // Print out the answer here
+	for (int i = 1; i <= n; i++) { cout << cnt[i] << " "; }
+	cout << endl;
+}
+```
+
+</CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class CoveredPointsCount {
+	public static void main(String[] args) throws IOException {
+		BufferedReader br =
+		    new BufferedReader(new InputStreamReader(System.in));
+		int n = Integer.valueOf(br.readLine());
```

Comment:
`Integer.valueOf`?

Reply:
emm yea should be `parseInt`, but somehow `valueOf` also works